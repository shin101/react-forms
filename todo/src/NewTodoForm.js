import React, {useState} from "react";
import { v4 as uuid } from "uuid";


const NewTodoForm = ({ addTask }) => {
  const initialState = {name: ''};

  const [formData, setFormData] = useState(initialState);
  const handleChange = (e) => {
    const {name, value} = e.target;
    setFormData(formData => ({
      ...formData,
      [name]: value
    }))
  }


  const handleSubmit = (e) => {
    e.preventDefault();
    addTask({...formData, id: uuid()});
    setFormData(initialState);

  }

  return (
    <form onSubmit={handleSubmit}>
      <label htmlFor="name">Tasks</label>
      <input 
      name="name" 
      placeholder="task name here" 
      value={formData.name}
      onChange={handleChange}
      />
      <button>Add Task</button>
    </form>
  )


}

export default NewTodoForm;