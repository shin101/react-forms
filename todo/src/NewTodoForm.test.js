import { render, fireEvent, queryByText } from "@testing-library/react";
import React from "react";
import NewTodoForm from "./NewTodoForm";

// smoke test
it("renders without crashing", function(){
  render(<NewTodoForm />);
});

// snapshot test
it("matches snapshot", function (){
  const { asFragment } = render(<NewTodoForm />);
  expect(asFragment()).toMatchSnapshot();
})

// it("should add a new task", function(){
//   const {getByLabelText} = render(<NewTodoForm />);
//   const input = getByLabelText("Tasks"); // if this doesnt work replace name with tasks 
//   const btn = queryByText("Add Task");
//   fireEvent.change(input, {target: { value: 'go to the dentist'}});
//   fireEvent.click(btn);
//   expect(queryByText('go to the dentist')).toBeInTheDocument();

// })

