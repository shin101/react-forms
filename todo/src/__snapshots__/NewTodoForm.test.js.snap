// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="name"
    >
      Tasks
    </label>
    <input
      name="name"
      placeholder="task name here"
      value=""
    />
    <button>
      Add Task
    </button>
  </form>
</DocumentFragment>
`;
